controllers/drcluster_controller.go

```diff
@@ -143,11 +146,20 @@ func (r *DRClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 }
 
 func (u *drclusterInstance) initializeStatus() {
-	// TODO: Only initialize those conditions whose status is
-	//       not available.
-	// Set the DRCluster conditions to unknown as nothing is known at this point
-	msg := "Initializing DRCluster"
-	setDRClusterInitialCondition(&u.object.Status.Conditions, u.object.Generation, msg)
+	// Save a copy of the instance status to be used for the DRCluster status update comparison
+	u.object.Status.DeepCopyInto(&u.savedInstanceStatus)
+
+	if u.savedInstanceStatus.Conditions == nil {
+		u.savedInstanceStatus.Conditions = []metav1.Condition{}
+	}
+
+	if u.object.Status.Conditions == nil {
```

Comment:
Raghu, Could you pls explain what is the difference between the if in 152 and 156. Because both are checking drcluster status condition.